src/detect-filestore.c

```diff
@@ -207,11 +235,33 @@ static int DetectFilestorePostMatch(DetectEngineThreadCtx *det_ctx,
 {
     SCEnter();
 
+    DEBUG_VALIDATE_BUG_ON(p->flow == NULL);
+
     if (det_ctx->filestore_cnt == 0) {
```

Comment:
By the way, `det_ctx->filestore_cnt` looks fishy to me cf https://github.com/OISF/suricata-verify/pull/1999

Its scope is one packet when a signature can take multiple packets to match...